src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs

```diff
@@ -22,4 +22,23 @@ public static class SiteServiceExtensions
     /// <returns>An instance of the given type if one exists.</returns>
     public static async Task<T> GetSettingsAsync<T>(this ISiteService siteService, string name) where T : new()
         => (await siteService.GetSiteSettingsAsync()).As<T>(name);
+
+    /// <summary>
+    /// Gets the site settings from the cache.
+    /// </summary>
+    public static ISite GetSiteSettings(this ISiteService siteService)
```

Comment:
Add/complete with a comment mentioning it's using the event handler to eagerly load the settings such that reading the settings synchronously is fine here. 
Also add a docs comment, more visible, that users should only use it in places that are not async, or they should use the other method.

Same for the `<TSettings>` version below.